### PR TITLE
Vagrant RPM

### PR DESCRIPTION
**Pull request by djosborne**
Switched vagrantfile to install RPMs instead of using the calico-mesos docker image.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -8,7 +8,8 @@ num_instances = 2
 instance_name_prefix="calico-mesos"
 calico_node_ver = "v0.17.0"
 calicoctl_url = "https://github.com/projectcalico/calico-containers/releases/download/#{calico_node_ver}/calicoctl"
-
+mesos_netmodules_rpm_url = "https://github.com/projectcalico/calico-mesos-deployments/releases/download/0.27.1%2B1/mesos-netmodules-rpms.tar"
+calico_mesos_rpm_url = "https://github.com/projectcalico/calico-mesos-deployments/releases/download/0.27.1%2B2/calico-mesos.rpm"
```
Comment (robbrockbank) [inline Vagrantfile]: Why different versions for each of these?

File: Vagrantfile [hunk 2]
```diff
@@ -62,89 +63,56 @@ Vagrant.configure("2") do |config|
...
-        host.vm.provision :shell, inline: "systemctl start mesos-master.service", privileged: true
-
-        # Marathon
-        host.vm.provision :shell, inline: "mv marathon.service /usr/lib/systemd/system/", privileged: true
-        host.vm.provision :shell, inline: "systemctl enable marathon.service", privileged: true
-        host.vm.provision :shell, inline: "systemctl start marathon.service", privileged: true
       end
 
-      # Configure the Agent nodes of the cluster.
+      # All the single agents, all the single agents
```
Comment (robbrockbank) [inline Vagrantfile]: hmmm,  really turkyonce?

File: Vagrantfile [hunk 3]
```diff
@@ -62,89 +63,56 @@ Vagrant.configure("2") do |config|
...
       if i > 1
-        ["calico", "mesos-agent"].each do |service_name|
-          host.vm.provision "file", source: "units/#{service_name}.service", destination: "#{service_name}.service"
+        host.vm.provision :docker, images: ["calico/node:#{calico_node_ver}"]
```
Comment (robbrockbank) [inline Vagrantfile]: And the lib network image too

File: Vagrantfile [hunk 4]
```diff
@@ -62,89 +63,56 @@ Vagrant.configure("2") do |config|
...
-          host.vm.provision :shell, inline: "firewall-cmd --zone=public --add-port=#{port}/tcp --permanent", privileged: true
+        host.vm.provision :shell, inline: "yum install -y calico-mesos.rpm", privileged: true
+      
+        # Configure calico
+        host.vm.provision :shell, inline: "sh -c 'echo MASTER=zk://#{master_ip}:2181/mesos/ > /etc/default/mesos-slave'", privileged: true
+        host.vm.provision :shell, inline: "sh -c 'echo ETCD_AUTHORITY=#{master_ip}:2379 >> /etc/default/mesos-slave'", privileged: true
+        host.vm.provision :shell, inline: "systemctl start calico-mesos.service", privileged: true
+
+        # Download and install mesos-netmodules-rpms.tar
+        if ENV["USE_LOCAL_MESOS_NETMODULES_TAR"] == "true"
```
Comment (robbrockbank) [inline Vagrantfile]: These environment options should be documented somewhere.

Comment (robbrockbank) [inline Vagrantfile]: Should these really be booleans, or should they be local filenames?

File: Vagrantfile (hunk 3 above)
Comment (robbrockbank) [inline Vagrantfile]: Ah nope - so lib network is not part of this.  Okey dokes.

File: Vagrantfile [hunk 5]
```diff
@@ -8,7 +8,8 @@ num_instances = 2
 instance_name_prefix="calico-mesos"
```
Comment (robbrockbank) [inline Vagrantfile]: I'd like  to see some (more) comments in this file.  We should be commenting the global variables, the magic environment variables, comment for each service that we are installing.

File: docs/DockerizedVagrant.md [hunk 6]
```diff
@@ -6,32 +6,31 @@
 > You are viewing the calico-mesos-deployments documentation for release **release**.
 <!--- end of master only -->
 
-# Deploying a Vagrant Dockerized Mesos Cluster with Calico
-This guide will start a running Mesos cluster with Calico Networking using a simple `vagrant up`. Note: This guide serves as a quick demo, but is not recommended for production use as it creates a Mesos Master and Agent on the same hypervisor.
-
-If you are looking for a guide to set up a POC in your lab, see the [Manual Dockerized Deployment guide](DockerizedDeployment.md).
+# Vagrant Deployed Mesos Cluster with Calico
+This guide will start a running Mesos cluster with Calico Networking using a simple `vagrant up`. If you are looking for a guide to perform the steps as this demo does, see the [RPM Guide](RpmInstallCalicoMesos.md).
```
Comment (robbrockbank) [inline docs/DockerizedVagrant.md]: I'd probably reword the second sentence to just say something like "The vagrant installation is based on the same installation procedure documented in out RPM Installation Guide." - that way the focus is more on what this guide does.

File: docs/DockerizedVagrant.md [hunk 7]
```diff
@@ -6,32 +6,31 @@
...
+   * `mesos-master`
+   * `etcd`
+   * `zookeeper`
+   * `marathon`
 
 ### Agent
+The Mesos Agent is configured with several custom RPMs. Mesos is installed with Netmodules using the Netmodules RPM bundle, and Calico is added and configured using the calico-mesos.rpm provided in this repository. Once both are installed, the agent will match the following specification:
  * **OS**: `Centos`
  * **Hostname**: `calico-02`
  * **IP**: `172.18.8.102`
```
Comment (robbrockbank) [inline docs/DockerizedVagrant.md]: Don't we support >1 agent?

File: docs/DockerizedVagrant.md [hunk 8]
```diff
@@ -6,32 +6,31 @@
...
  * **Docker Containers**:
-	 * `mesos-agent` - (`calico/mesos-calico`)
-	 * `calico-node` - (`calico/node`)
+   * `mesos-agent`
```
Comment (robbrockbank) [inline docs/DockerizedVagrant.md]: Is the Mesos Agent running as a container?

File: docs/RpmInstallCalicoMesos.md [hunk 9]
```diff
@@ -7,11 +7,8 @@
 <!--- end of master only -->
 
 # RPM Install Calico-Mesos
-This tutorial will walk through installing prebuilt RPMs containg Mesos, Netmodules, and Calico. At the completion of this guide, you will have a Mesos Slave ready to launch tasks with Calico networking.
-
->Note: These RPMs do not serve as an official calico or mesos installation option, and will not receive a supported upgrade story for the future. They merely serve as a more automatic alternative to performing the steps in the [Manually Installing Calico + Mesos + Netmodules](ManualInstallCalicoMesos.md) guide. 
-
-This RPM installation includes and installs:
+This tutorial shows how to configure a Mesos Agent able to launch tasks networked with calico networking.
+At the completion of this guide, you will have a Mesos Slave ready to launch tasks with their own Calico assigned IP Addresses using the following services and configuration files:
```
Comment (robbrockbank) [inline docs/RpmInstallCalicoMesos.md]: Slave->Agent

File: docs/RpmInstallCalicoMesos.md [hunk 10]
```diff
@@ -7,11 +7,8 @@
 <!--- end of master only -->
 
 # RPM Install Calico-Mesos
```
Comment (robbrockbank) [inline docs/RpmInstallCalicoMesos.md]: Can we beef up the title:   RPM installation of Mesos and Marathon with Calico  (or similar)

File: docs/RpmInstallCalicoMesos.md [hunk 11]
```diff
@@ -7,11 +7,8 @@
...
-This tutorial will walk through installing prebuilt RPMs containg Mesos, Netmodules, and Calico. At the completion of this guide, you will have a Mesos Slave ready to launch tasks with Calico networking.
-
->Note: These RPMs do not serve as an official calico or mesos installation option, and will not receive a supported upgrade story for the future. They merely serve as a more automatic alternative to performing the steps in the [Manually Installing Calico + Mesos + Netmodules](ManualInstallCalicoMesos.md) guide. 
-
-This RPM installation includes and installs:
+This tutorial shows how to configure a Mesos Agent able to launch tasks networked with calico networking.
```
Comment (robbrockbank) [inline docs/RpmInstallCalicoMesos.md]: Should mention Marathon in here too.  Maybe blurb that Marathon is not required yada yada yada.

File: docs/DockerizedVagrant.md (hunk 8 above)
Comment (djosborne) [inline docs/DockerizedVagrant.md]: No, its RPM installed directly onto the host

File: docs/DockerizedVagrant.md (hunk 7 above)
Comment (djosborne) [inline docs/DockerizedVagrant.md]: There's a section at the end of this doc that explains increasing the number of agents, and what their IPs will be

File: Vagrantfile (hunk 1 above)
Comment (djosborne) [inline Vagrantfile]: I wish. You can't use the "latest" links for downloading artifacts. They won't resolve. They just 404. You need to point directly at an artifact.

File: docs/RpmInstallCalicoMesos.md (hunk 11 above)
Comment (djosborne) [inline docs/RpmInstallCalicoMesos.md]: Can you elaborate on this? The Prerequisites section states that you should have Marathon running...

**Pull request merged**